test/test_nn.py

```diff
@@ -16536,6 +16536,44 @@ def forward_hook(m, input, output):
         expected_grad = -sig_x * (1 - sig_x) * 2 * mask
         self.assertEqual(input.grad, expected_grad)
 
+    def test_module_forward_preforward_hook_removable(self):
```

Comment:
Could you please add a comment or two for these tests describing what they're testing for posterity?